src/thruster.cpp

```diff
@@ -85,49 +83,34 @@ void Thruster::Load(physics::ModelPtr _parent, sdf::ElementPtr _sdf)
     ROS_DEBUG_STREAM("Thruster plugin initialized");
     ROS_DEBUG_STREAM("Thruster_settings: " << thruster_settings);
     ROS_DEBUG_STREAM("Thruster names: ");
-    for(int i=0; i<num_thrusters; i++)
+    for(unsigned int i=0; i<num_thrusters; i++)
     {
         // Get a pointer for each thruster link
         physics::LinkPtr t = sub->GetLink(thruster_names[i]);
         thruster_links.push_back(t);
         ROS_DEBUG_STREAM("t->GetRelativePose(): " << t->GetRelativePose());
+    }
 
-        // The visual message that will be sent to the gzserver
-        // will tell it to create a cylinder showing thruster output
-        // This initializes a message for each thruster
-        msgs::Visual v_msg;
-        visualMsg.push_back(v_msg);
-        visualMsg[i].set_name(thruster_names[i] + "_force_vis");
-        visualMsg[i].set_parent_name(thruster_names[i]);
-        visualMsg[i].set_cast_shadows(false);
+    ReloadParams();
 
-        // The geomMsg is part of the visual message
-        // It defines an object to visualize, in this case,
-        // a narrow cylinder
-        msgs::Geometry *geomMsg = visualMsg[i].mutable_geometry();
-        geomMsg->set_type(msgs::Geometry::CYLINDER);
-        geomMsg->mutable_cylinder()->set_radius(.004);
-        geomMsg->mutable_cylinder()->set_length(1);
-
-        // Set color to red
-        // TODO: They aren't really all that red...
-        msgs::Material *matMsg = visualMsg[i].mutable_material();
-        msgs::Set(matMsg->mutable_diffuse(), common::Color::Red);
-        msgs::Set(matMsg->mutable_specular(), common::Color::Red);
+    InitVisualizers();
 
-        // Set pose of line to somewhere in the distance initially
-        msgs::Set(visualMsg[i].mutable_pose(), ignition::math::Pose3d(1000, 1000, 1000, 0, 0, 0));
-
-        // Publish message
-        visPub->Publish(visualMsg[i]);
-    }
+    visualize_update_time = ros::Duration(0.2);
```

Comment:
Perhaps make this a parameter? Or at least a constant somewhere. Either way, please update as per @bkallaher's comment.